### ISSUE TITLE
Feature - Puppet Enterprise Database Access

### ISSUE THREAD

**Issue by MartyEwings** [puppetlabs/puppetlabs-puppet_operations_appliance#4]
Description

Allow read only access to all Puppet Enterprise database schemas on Primary and Replica Servers.

Purpose

To allow the support engineers to log onto the live Postgres databases on the primary and replica servers and run queries and data retrieval exercises

Assumptions

Command Line Access using a PSQL client

Acceptance criteria
Agent Certificate authenticated access
Dedicated database user and role for RSAN
“SELECT” level access to  pe-puppetdb,  pe-rbac,  pe-activity,  pe-orchestrator, pe-inventory,  pe-classifier databases on all configured pe-postgresql infrastructure nodes
Option to disable database access

**Comment by MartyEwings** [comment]
exporter class ill need something like this for all the pe databases... (example just puppetdb)


```
  if $rsan_host {
    $_rsan_host = $rsan_host
  } else {
    $_query = puppetdb_query('resources[certname] {
      type = "Class" and
      title = "Rsan" and
      nodes {
        deactivated is null and
        expired is null
      }
      order by certname asc
      limit 1
    }')
    unless $_query.empty {
      $_rsan_host = $_query[0]['certname']
    }
  }

  # If $rsan_host is not defined and the query fails to find a rsan  host, issue a warning.

  if $_rsan_host == undef {

    notify { 'You must specify rsan_host (or apply the rsan class to an agent) to enable access.': }

  } else {

    pe_postgresql::server::role { 'rsan': }

    pe_postgresql::server::database_grant { 'rsan':
      privilege => 'SELECT',
      db        => 'pe-puppetdb',
      role      => 'rsan',
    }

    # If the fact doesn't exist then PostgreSQL is probably version 9.4.

    if $facts['pe_postgresql_info']['installed_server_version'] {
      $postgres_version = $facts['pe_postgresql_info']['installed_server_version']
    } else {
      $postgres_version = '9.4'
    }

    puppet_enterprise::pg::cert_whitelist_entry { 'allow-rsan-access':
      user                          => 'rsan',
      database                      => 'pe-puppetdb',
      allowed_client_certname       => $_rsan_host,
      pg_ident_conf_path            => "/opt/puppetlabs/server/data/postgresql/${postgres_version}/data/pg_ident.conf",
      ip_mask_allow_all_users_ssl   => '0.0.0.0/0',
      ipv6_mask_allow_all_users_ssl => '::/0',
    }
```

SELECT status is important

**Comment by MartyEwings** [comment]
importer class will need only :

include postgresql::client

**Comment by kreeuwijk** [comment]
Fixed in #11
